homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,413 @@
+                    self._devicetype,
+                )
+            await self._controller.connect()
+            self._connection_retries += 1
+        else:
+            if not self._connected:
+                self._connected = True
+                _LOGGER.debug("Restored connection to %s API.", self._devicetype)
+            self._connection_retries = 0
+
+        self._current_temp = self._controller.get_temperature(self._deviceid)
+        self._fan_speed = self._controller.get_fan_speed(self._deviceid)
+        self._power = self._controller.is_on(self._deviceid)
+        self._min_temp = self._controller.get_min_setpoint(self._deviceid)
+        self._max_temp = self._controller.get_max_setpoint(self._deviceid)
+        self._rssi = self._controller.get_rssi(self._deviceid)
+        self._run_hours = self._controller.get_run_hours(self._deviceid)
+        self._target_temp = self._controller.get_setpoint(self._deviceid)
+        mode = self._controller.get_mode(self._deviceid)
+
+        # Operation mode
+        self._hvac_mode = MAP_OPERATION_MODE.get(mode, STATE_UNKNOWN)
+
+        # Swing mode
+        # Climate module only supports one swing setting.
+        if self._has_swing_control:
+            self._vvane = self._controller.get_vertical_swing(self._deviceid)
+            self._hvane = self._controller.get_horizontal_swing(self._deviceid)
+
+            if self._vvane == "swing" and self._hvane == "swing":
+                self._swing = SWING_BOTH
+            elif self._vvane == "swing":
+                self._swing = SWING_VERTICAL
+            elif self._hvane == "swing":
+                self._swing = SWING_HORIZONTAL
+            else:
+                self._swing = SWING_OFF
+
+    async def async_will_remove_from_hass(self):
+        """Shutdown the controller when the device is being removed."""
+        await self._controller.stop()
+
+    @property
+    def icon(self):
+        """Return the icon for the current state."""
+        icon = None
+        if self._power:
+            icon = MAP_STATE_ICONS.get(self._hvac_mode)
+        return icon
+
+    def update_callback(self):
+        """Let HA know there has been an update from the controller."""
+        _LOGGER.debug("%s API sent a status update.", self._devicetype)
+        self.schedule_update_ha_state(True)
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature for the current mode of operation."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature for the current mode of operation."""
+        return self._max_temp
+
+    @property
+    def should_poll(self):
+        """Poll for updates if pyIntesisHome doesn't have a socket open."""
+        # This could be switched on controller.is_connected, but HA doesn't
+        # seem to handle dynamically changing from push to poll.
+        return True
```

Comment:
This will only be checked once when the entity is added. We don't allow dynamically changing it.